lib/modules/manager/vendir/artifacts.spec.ts

```diff
@@ -0,0 +1,356 @@
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache',
+    );
+    fs.getParentDir.mockReturnValue('');
+
+    // artifacts
+    fs.getSiblingFileName.mockReturnValueOnce('vendor');
+    git.getRepoStatus.mockResolvedValueOnce(
+      partial<StatusResult>({
+        not_added: ['vendor/Chart.yaml'],
+        deleted: ['vendor/removed.yaml'],
+      }),
+    );
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: vendirFile,
+        config: {
+          postUpdateOptions: ['vendirUpdateSubChartArchives'],
+          ...config,
+        },
+      }),
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'vendor/Chart.yaml',
+          contents: undefined,
+        },
+      },
+      {
+        file: {
+          type: 'deletion',
+          path: 'vendor/removed.yaml',
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'vendir sync',
+        options: {
+          env: {
+            HOME: '/home/user',
+            HTTPS_PROXY: 'https://example.com',
+            HTTP_PROXY: 'http://example.com',
+            LANG: 'en_US.UTF-8',
+            LC_ALL: 'en_US',
+            NO_PROXY: 'localhost',
+            PATH: '/tmp/path',
+          },
+        },
+      },
+    ]);
+  });
+
+  it('add artifacts without old archives', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(null);
+    fs.getSiblingFileName.mockReturnValueOnce('vendir.yml');
+    const execSnapshots = mockExecAll();
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache',
+    );
+    fs.getParentDir.mockReturnValue('');
+
+    // artifacts
+    fs.getSiblingFileName.mockReturnValueOnce('vendor');
+    git.getRepoStatus.mockResolvedValueOnce(
+      partial<StatusResult>({
+        not_added: ['vendor/Chart.yaml'],
+      }),
+    );
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: vendirFile,
+        config: {
+          postUpdateOptions: ['vendirUpdateSubChartArchives'],
+          ...config,
+        },
+      }),
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'vendor/Chart.yaml',
+          contents: undefined,
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'vendir sync',
+        options: {
+          env: {
+            HOME: '/home/user',
+            HTTPS_PROXY: 'https://example.com',
+            HTTP_PROXY: 'http://example.com',
+            LANG: 'en_US.UTF-8',
+            LC_ALL: 'en_US',
+            NO_PROXY: 'localhost',
+            PATH: '/tmp/path',
+          },
+        },
+      },
+    ]);
+  });
```

Comment:
add tests for binary source with install and docker mode

Reply:
Done!